server/macros/currency_base.py

```diff
@@ -0,0 +1,42 @@
+# -*- coding: utf-8; -*-
+#
+# This file is part of Superdesk.
+#
+# Copyright 2013, 2014 Sourcefabric z.u. and contributors.
+#
+# For the full copyright and license information, please see the
+# AUTHORS and LICENSE files distributed with this source code, or
+# at https://www.sourcefabric.org/superdesk/license
+
+import requests
+import re
+
+RATE_SERVICE = 'http://download.finance.yahoo.com/d/quotes.csv?s={}=X&f=nl1d1'
+
+
+def get_rate(from_currency, to_currency):
+    """Get the exchange rate."""
+    r = requests.get(RATE_SERVICE.format(from_currency + to_currency), timeout=5)
+    return float(r.text.split(',')[1])
+
+
+def do_conversion(item, rate, prefix, search_param):
+    """
+    Performs the conversion
+    :param item: story
+    :param rate: exchange rate
+    :param prefix: to be used in the results
+    :param search_param: search parameter to locate the original value
```

Comment:
IMO it should be made a bit more clear that `search_param` should, in fact, be a valid regular expression pattern, and not just an arbitrary string, for example.